src/Stratis.FederatedPeg.Tests/TransactionBuilderTests.cs

```diff
@@ -0,0 +1,169 @@
+﻿using System.Collections.Generic;
+using System.Text;
+using Microsoft.Extensions.Logging;
+using Moq;
+using NBitcoin;
+using NSubstitute;
+using Stratis.Bitcoin.Configuration;
+using Stratis.Bitcoin.Features.Wallet.Interfaces;
+using Stratis.Bitcoin.Utilities;
+using Stratis.FederatedPeg.Features.FederationGateway;
+using Stratis.FederatedPeg.Features.FederationGateway.Interfaces;
+using Stratis.FederatedPeg.Features.FederationGateway.Wallet;
+using Stratis.Sidechains.Networks;
+using Xunit;
+using Recipient = Stratis.FederatedPeg.Features.FederationGateway.Wallet;
+
+namespace Stratis.FederatedPeg.Tests
+{
+    /// <summary>
+    /// Tests whether transactions are built deterministically.
+    /// </summary>
+    public class TransactionBuilderTests
+    {
+        private Network network;
+        private IDateTimeProvider dateTimeProvider;
+        private ILoggerFactory loggerFactory;
+        private IFederationGatewaySettings settings;
+        private IFederationWalletManager federationWalletManager;
+        private IFederationWalletManager federationWalletManager2;
+        private IFederationWalletTransactionHandler federationTransactionHandler;
+        private ConcurrentChain chain;
+        private NodeSettings nodeSettings;
+        private DataFolder dataFolder;
+        private IWalletFeePolicy walletFeePolicy;
+        private FederationWallet wallet;
+
+        public TransactionBuilderTests()
+        {
+            this.network = ApexNetwork.RegTest;
+            this.nodeSettings = new NodeSettings(this.network, NBitcoin.Protocol.ProtocolVersion.ALT_PROTOCOL_VERSION,
+                args: new[] {
+                    "-redeemscript=2 026ebcbf6bfe7ce1d957adbef8ab2b66c788656f35896a170257d6838bda70b95c 02a97b7d0fad7ea10f456311dcd496ae9293952d4c5f2ebdfc32624195fde14687 02e9d3cd0c2fa501957149ff9d21150f3901e6ece0e3fe3007f2372720c84e3ee1 03c99f997ed71c7f92cf532175cea933f2f11bf08f1521d25eb3cc9b8729af8bf4 034b191e3b3107b71d1373e840c5bf23098b55a355ca959b968993f5dec699fc38 5 OP_CHECKMULTISIG",
+                    "-publickey=026ebcbf6bfe7ce1d957adbef8ab2b66c788656f35896a170257d6838bda70b95c"
+                });
+            this.loggerFactory = Substitute.For<ILoggerFactory>();
+            this.settings = new FederationGatewaySettings(this.nodeSettings);
+            this.chain = new ConcurrentChain(this.network);
+            this.dataFolder = this.nodeSettings.DataFolder;
+            var mockWalletFeePolicy = new Mock<IWalletFeePolicy>();
+            this.walletFeePolicy = mockWalletFeePolicy.Object;
+            this.dateTimeProvider = DateTimeProvider.Default;
+
```

Comment:
So in the end did you prefer Moq or NSubstitute ? :)

Reply:
I am more familiar with Moq and I see it gets used a lot on the FullNode. :)